armi/reactor/converters/axialExpansionChanger/assemblyAxialLinkage.py

```diff
@@ -183,31 +268,50 @@ def _getLinkedComponents(self, b, c):
         RuntimeError
             multiple candidate components are found to be axially linked to a component
         """
-        lstLinkedC = [None, None]
-        for ib, linkdBlk in enumerate(self.linkedBlocks[b]):
-            if linkdBlk is not None:
-                for otherC in getSolidComponents(linkdBlk.getChildren()):
-                    if _determineLinked(c, otherC):
-                        if lstLinkedC[ib] is not None:
-                            errMsg = (
-                                "Multiple component axial linkages have been found for "
-                                f"Component {c}; Block {b}; Assembly {b.parent}."
-                                " This is indicative of an error in the blueprints! Linked "
-                                f"components found are {lstLinkedC[ib]} and {otherC}"
-                            )
-                            runLog.error(msg=errMsg)
-                            raise RuntimeError(errMsg)
-                        lstLinkedC[ib] = otherC
-
+        linkedBlocks = self.linkedBlocks[b]
+        lowerC = self._findComponentLinkedTo(c, linkedBlocks.lower)
+        upperC = self._findComponentLinkedTo(c, linkedBlocks.upper)
+        lstLinkedC = AxialLink(lowerC, upperC)
         self.linkedComponents[c] = lstLinkedC
 
-        if lstLinkedC[0] is None:
+        if self.linkedBlocks[b].lower is None and lstLinkedC.lower is None:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked below it!",
                 single=True,
             )
-        if lstLinkedC[1] is None:
+        if self.linkedBlocks[b].upper is None and lstLinkedC.upper is None:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked above it!",
                 single=True,
             )
+
+    @staticmethod
+    def areAxiallyLinked(componentA: Component, componentB: Component) -> bool:
+        """Check if two components are axially linked.
+
+        Components are considered linked if the following are found to be true:
```

Comment:
This and the following list feel like they can be deleted since they're restated up in in the `areAxiallyLinked` function.

Reply:
Yeah I can remove the checklist here since we reference it in the other function.